src/cbc_sdk/platform/devices.py

```diff
@@ -616,13 +759,55 @@ def _run_async_query(self, context):
             output += [self._doc_class(self._cb, item["id"], item) for item in results]
         return output
 
+    def facets(self, fieldlist, max_rows=0):
+        """
+        Return information about the facets for all known evices, using the defined criteria.
+
+        Example:
+            >>> query = api.select(Device).where('')
+            >>> facets = query.facets(['policy_id', 'status', 'os', 'ad_group_id'])
+            >>> for f in facets:
+            ...     print(f"Field {f.field} - {len(f.values_)} distinct values")
+
+        Required Permissions:
+            device (READ)
+
+        Args:
+            fieldlist (list[str]): List of facet field names. Valid names are "policy_id", "status", "os",
+                                   and "ad_group_id".
+            max_rows (int): The maximum number of rows to return. 0 means return all rows.
+
+        Returns:
+            list[DeviceFacet]: A list of facet information.
+        """
+        if not fieldlist:
+            raise ApiError("At least one term field must be specified")
+        if not all((field in DeviceSearchQuery.VALID_FACET_FIELDS) for field in fieldlist):
```

Comment:
I understand the need to protect against the potential API error however this would block if the backend API supports additional facet fields. I believe we can rely on the API error to provide this feedback instead of building it in client side

Reply:
Also, other "facet" APIs like for USB devices use client-side validation of the facet fields in the same fashion.

Reply:
Given the change in field names this doesn't make sense